Model/Token/RequestData.php

```diff
@@ -311,18 +294,34 @@ protected function itemArgs($order)
                 $items[] = [
                     'title' => $item->getName(),
                     'code' => $item->getSku(),
-                    'amount' => floatval($item->getQtyOrdered()),
+                    'amount' => $item->getQtyOrdered(),
                     'price' => 0,
                     'vat' => 0
                 ];
             } else {
-                $items[] = [
+                $paytrailItem = [
                     'title' => $item->getName(),
                     'code' => $item->getSku(),
-                    'amount' => floatval($item->getQtyOrdered()),
-                    'price' => floatval($item->getPriceInclTax()) - round(($discountIncl / $item->getQtyOrdered()), 2),
-                    'vat' => round(floatval($item->getTaxPercent()))
+                    'amount' => $item->getQtyOrdered(),
+                    'price' => $item->getPriceInclTax() - ($discountIncl / $item->getQtyOrdered()),
```

Comment:
Is $item->getQtyOrdered() always > 0 so that no division by zero occur? There is also bcdiv() function altho not sure if there is need to use that.